apps/sens/fsw/unit_test/sens_cmds_test.cpp

```diff
@@ -48,43 +51,239 @@
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
+
+    sprintf(expectedEvent, "Recvd invalid command code (%u)", 100);
+
+    UtAssert_EventSent(SENS_CC_ERR_EID, CFE_EVS_ERROR, expectedEvent,
+                       "ProcessCmdPipe, InvalidCmd Event Sent");
+    UtAssert_True(oSENS.HkTlm.usCmdErrCnt == 1, "ProcessCmdPipe, InvalidCmd");
+}
+
+
+/**
+ * Test SENS ProcessCmdPipe, CmdPipeError
+ */
+void Test_SENS_ProcessCmdPipe_CmdPipeError(void)
+{
+    SENS              oSENS;
+
+    int32             SchPipe;
+    SENS_NoArgCmd_t   InMsg;
+    char              expectedEvent[CFE_EVS_MAX_MESSAGE_LENGTH];
+
+    /* The following will emulate the behavior of receiving a message,
+       and gives it data to process. */
+    SchPipe = Ut_CFE_SB_CreatePipe("SENS_SCH_PIPE");
+    CFE_SB_InitMsg ((void*)&InMsg, SENS_SEND_HK_MID, sizeof(InMsg), TRUE);
+    CFE_SB_SetCmdCode ((CFE_SB_MsgPtr_t)&InMsg, (uint16)0);
+    Ut_CFE_SB_AddMsgToPipe((void*)&InMsg, (CFE_SB_PipeId_t)SchPipe);
+
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SB_BAD_ARGUMENT, 2);
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
+
+    sprintf(expectedEvent, "CMD pipe read error (0x%08lX)", CFE_SB_BAD_ARGUMENT);
+
+    UtAssert_EventSent(SENS_RCVMSG_ERR_EID, CFE_EVS_ERROR, expectedEvent,
+                       "ProcessCmdPipe, CmdPipeError");
+}
+
+
+/**
+ * Test SENS ProcessCmdPipe, Noop
+ */
+void Test_SENS_ProcessCmdPipe_Noop(void)
 {
-    return 5;
+    SENS              oSENS;
+
+    int32             CmdPipe;
+    SENS_NoArgCmd_t   InMsg;
+    char              expectedEvent[CFE_EVS_MAX_MESSAGE_LENGTH];
+
+    /* The following will emulate the behavior of receiving a message,
+       and gives it data to process. */
+    CmdPipe = Ut_CFE_SB_CreatePipe("SENS_CMD_PIPE");
+    CFE_SB_InitMsg ((void*)&InMsg, SENS_CMD_MID, sizeof(InMsg), TRUE);
+    CFE_SB_SetCmdCode ((CFE_SB_MsgPtr_t)&InMsg, (uint16)SENS_NOOP_CC);
+    Ut_CFE_SB_AddMsgToPipe((void*)&InMsg, (CFE_SB_PipeId_t)CmdPipe);
+
+    SENS_Test_PrintCmdMsg((void*)&InMsg, sizeof(SENS_NoArgCmd_t));
+
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SUCCESS, 1);
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_GETMSGID_INDEX, SENS_SEND_HK_MID, 1);
+
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_RUNLOOP_INDEX, FALSE, 2);
+
+    /* Execute the function being tested */
+    oSENS.AppMain();
+
+    sprintf(expectedEvent, "Recvd NOOP. Version %d.%d.%d.%d",
+                            SENS_MAJOR_VERSION, SENS_MINOR_VERSION,
+                            SENS_REVISION, SENS_MISSION_REV);
+
+    UtAssert_EventSent(SENS_CMD_NOOP_EID, CFE_EVS_INFORMATION, expectedEvent,
+                       "ProcessCmdPipe, Noop Event Sent");
+    UtAssert_True(oSENS.HkTlm.usCmdCnt == 1, "ProcessCmdPipe, Noop");
```

Comment:
Excellent unit test.  This is perfect.  Great use of the self adapting design pattern, and you tested all the effects you could.